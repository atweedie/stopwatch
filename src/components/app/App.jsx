import '../../styles/App.css';
import { TimerState } from '../../modules/timerStateContext';
import TimerDisplay from '../timerDisplay/TimerDisplay';
import { useState } from 'react';
import TimerToggleButton from '../timerToggleButton/TimerToggleButton';

const App = () => {
    const [elapsedTime, setElapsedTime] = useState(0);
    const [isRunning, setIsRunning] = useState(false);

    return (
        <div className="App">
            <TimerState.Provider value={{elapsedTime, setElapsedTime, isRunning, setIsRunning}}>
                <TimerDisplay/>
                <TimerToggleButton/>
            </TimerState.Provider>
        </div>
    )
};

export default App;
