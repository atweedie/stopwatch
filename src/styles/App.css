.App {
  position: relative;
  top: 30vh;
  text-align: center;
}

.button {
  background-color: #0078ff;
  font-size: 1em;
  font-weight: bold;
  color: white;
  border: none;
  padding: 16px;
  margin: 16px;
  width: 10vw;
  min-width: 96px;
}

.button:hover {
  background-color: #0064e6;
}

.timer {
  font-size: 4em;
}