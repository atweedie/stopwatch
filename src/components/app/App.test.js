import { render } from '@testing-library/react';
import App from './App';

test('should render as expected', () => {
  const component = render(<App />);
 
  expect(component.container).toMatchSnapshot();
});
