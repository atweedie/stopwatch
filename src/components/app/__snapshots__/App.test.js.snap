// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<div>
  <div
    class="App"
  >
    <h1
      class="timer"
      id="stopwatch-elapsed"
    >
      0:00:00.00
    </h1>
    <button
      class="button"
      id="stopwatch-start-stop"
    >
      Start
    </button>
    <button
      class="button"
      id="stopwatch-reset"
    >
      Reset
    </button>
  </div>
</div>
`;
